File: gestion_equipements/src/main/java/models/Equipement.java
package models;

import java.time.LocalDate;

public class Equipement {
    private int id;
    private String name;
    private String description;
    private int categoryId;
    private float price;
    private String image;
    private boolean availability;
    private LocalDate dateAdded;
    private int partnerId;
    private String categoryName;

    public Equipement(int id, String name, String description, int categoryId, float price, String image, boolean availability, LocalDate dateAdded, int partnerId, String categoryName) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.categoryId = categoryId;
        this.price = price;
        this.image = image;
        this.availability = availability;
        this.dateAdded = dateAdded;
        this.partnerId = partnerId;
        this.categoryName = categoryName;
    }

    // Setters and Getters
    public int getId() { return id; }
    public String getName() { return name; }
    public String getDescription() { return description; }
    public int getCategoryId() { return categoryId; }
    public float getPrice() { return price; }
    public String getImage() { return image; }
    public boolean getAvailability() { return availability; }
    public LocalDate getDateAdded() { return dateAdded; }
    public int getPartnerId() { return partnerId; }
    public String getCategoryName() { return categoryName; }

    public void setName(String name) { this.name = name; }
    public void setDescription(String description) { this.description = description; }
    public void setCategoryId(int categoryId) { this.categoryId = categoryId; }
    public void setPrice(float price) { this.price = price; }
    public void setImage(String image) { this.image = image; }
    public void setAvailability(boolean availability) { this.availability = availability; }
    public void setDateAdded(LocalDate dateAdded) { this.dateAdded = dateAdded; }
    public void setPartnerId(int partnerId) { this.partnerId = partnerId; }
}

File: src/Entity/Equipement.php
<?php

namespace App\Entity;

use App\Repository\EquipementRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: EquipementRepository::class)]
class Equipement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le nom de l'équipement ne peut pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "Le nom ne peut pas dépasser 255 caractères.")]
    private ?string $name = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "La description de l'équipement ne peut pas être vide.")]
    #[Assert\Length(max: 500, maxMessage: "La description ne peut pas dépasser 500 caractères.")]
    private ?string $description = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false)]
    #[Assert\NotNull(message: "Veuillez sélectionner une catégorie.")]
    private ?Category $category = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: 'Le prix ne peut pas être vide.')]
    #[Assert\Type(type: 'float', message: 'Le prix doit être un nombre décimal.')]
    #[Assert\GreaterThanOrEqual(value: 0, message: 'Le prix ne peut pas être négatif.')]
    private ?float $price = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "L'image ne peut pas être vide.")]
    #[Assert\Url(message: "L'URL de l'image doit être valide.")]
    private ?string $image = null;

    #[ORM\Column]
    private ?bool $availability = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    #[Assert\NotBlank(message: "La date d'ajout ne peut pas être vide.")]
    #[Assert\Type("\DateTimeInterface", message: "La date d'ajout doit être valide.")]
    private ?\DateTimeInterface $dateAdded = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: true)]
    private ?User $partner = null;

    public function __construct()
    {
        $this->dateAdded = new \DateTime(); // Date actuelle par défaut
        $this->availability = true;         // Disponible par défaut
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): static
    {
        $this->name = $name;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getCategory(): ?Category
    {
        return $this->category;
    }

    public function setCategory(?Category $category): static
    {
        $this->category = $category;
        return $this;
    }

    public function getPrice(): ?float
    {
        return $this->price;
    }

    public function setPrice(float $price): static
    {
        $this->price = $price;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): static
    {
        $this->image = $image;
        return $this;
    }

    public function isAvailability(): ?bool
    {
        return $this->availability;
    }

    public function setAvailability(bool $availability): static
    {
        $this->availability = $availability;
        return $this;
    }

    public function getDateAdded(): ?\DateTimeInterface
    {
        return $this->dateAdded;
    }

    public function setDateAdded(\DateTimeInterface $dateAdded): static
    {
        $this->dateAdded = $dateAdded;
        return $this;
    }

    public function getPartner(): ?User
    {
        return $this->partner;
    }

    public function setPartner(?User $partner): static
    {
        $this->partner = $partner;
        return $this;
    }
}
